docs/source/public_api/kernels.rst

```diff
@@ -0,0 +1,12 @@
+probnum.kernels
+========================
+
+Kernels or covariance functions.
+
+Kernels describes the spatial or temporal variation of a random process. If
+evaluated at two sets of points a kernel is defined as the covariance of the
+values of the random process at these locations.
```

Comment:
Can you imagine any use case that is not a Gaussian processes? If not, maybe change this text here accordingly?

Reply:
Student-t processes come to mind. You can define a covariance function for arbitrary random processes. Maybe this is an argument for naming the class `CovarianceFunction`. This would also nicely correspond to a `Covariance` class for random variables, which might be useful in the future.